lib/kernel/src/dist_util.erl

```diff
@@ -695,8 +695,14 @@ send_status(#hs_data{socket = Socket, other_node = Node,
 	_ -> 
 	    true
     end.
-    
-    
+
+%% Constant time comparison between two binaries of the same size.
+secure_compare(B1, B2) ->
+    secure_compare(B1, B2, 0) =:= 0.
+secure_compare(<<H1, T1/binary>>, <<H2, T2/binary>>, Acc) ->
+    secure_compare(T1, T2, Acc bor (H1 bxor H2));
+secure_compare(<<>>, <<>>, Acc) ->
+    Acc.
```

Comment:
We should not assume the two binaries are the same size here, since the md5 usage may change in the future, and different versions may need to use different hashes.  When that happens, it is best to keep this comparison source code the same.  An example of a time-insensitive comparison with any length is at https://github.com/CloudI/CloudI/blob/develop/src/lib/cloudi_core/src/cloudi_string.erl#L415-L422 .

You may argue that any change in hash algorithm is handled elsewhere, but as source code becomes deprecated, its possible that the source code won't catch everything, and checking the length first would defeat the goal of this function.